File: src/types/tag.ts
export interface Tag {
  id: string;
  project_id: string;
  name: string;
  color: string | null;
}

export interface TagWithUsageCount {
  id: string;
  project_id: string;
  name: string;
  color: string | null;
  usage_count: number;
}

export interface CreateTagInput {
  project_id: string;
  name: string;
  color?: string | null;
}

export interface SetEntityTagsInput {
  project_id: string;
  entity_type: 'character' | 'location' | 'worldbuilding' | 'timeline' | 'note' | 'manuscript';
  entity_id: string;
  tag_names: string[];
}

File: src-tauri/src/models/tag.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Tag {
    pub id: String,
    pub project_id: String,
    pub name: String,
    pub color: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EntityTag {
    pub id: String,
    pub tag_id: String,
    pub entity_type: String,
    pub entity_id: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TagWithUsageCount {
    pub id: String,
    pub project_id: String,
    pub name: String,
    pub color: Option<String>,
    pub usage_count: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateTagInput {
    pub project_id: String,
    pub name: String,
    pub color: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SetEntityTagsInput {
    pub project_id: String,
    pub entity_type: String,
    pub entity_id: String,
    pub tag_names: Vec<String>,
}
